src/Mapping/Driver/ReflectionBasedDriver.php

```diff
@@ -32,8 +32,13 @@ private function isRepeatedPropertyDeclaration(ReflectionProperty $property, Cla
                 || $metadata->isInheritedEmbeddedClass($property->name);
         }
 
+        /** @var class-string $declaringClass */
         $declaringClass = $property->class;
 
+        if ($this->isTransient($declaringClass)) {
```

Comment:
One more issue here: This assumes that the same mapping driver that is working on `$property` can also judge whether `$declaringClass` is transient.

This may not work out e. g. when using different mapping drivers, for example mixing XML and attribute based configuration in different namespaces.